x-pack/plugins/maps/public/kibana_services.ts

```diff
@@ -63,23 +59,27 @@ export const getMapAppConfig = () => mapAppConfig;
 export const getShowMapsInspectorAdapter = () => getMapAppConfig().showMapsInspectorAdapter;
 export const getPreserveDrawingBuffer = () => getMapAppConfig().preserveDrawingBuffer;
 
-// map.* kibana.yml settings from maps_ems plugin that are shared between OSS map visualizations and maps app
-let kibanaCommonConfig: MapsEmsConfig;
-export const setKibanaCommonConfig = (config: MapsEmsConfig) => (kibanaCommonConfig = config);
-export const getKibanaCommonConfig = () => kibanaCommonConfig;
-
+let mapsEms: MapsEmsPluginPublicStart;
 let emsSettings: EMSSettings;
-export const setEMSSettings = (value: EMSSettings) => {
-  emsSettings = value;
+export const setMapsEmsStart = (value: MapsEmsPluginPublicStart) => {
+  mapsEms = value;
+  emsSettings = mapsEms.createEMSSettings();
 };
-export const getEMSSettings = () => {
+// map.* kibana.yml settings from maps_ems plugin that are shared between OSS map visualizations and maps app
```

Comment:
You can remove this comment as it does not really align with getMapsEmsStart. Maybe this comment can be moved to MapConfig type definition in mapsEms plugin